File: web/src/main/java/me/robinvdb/runetracker/database/DataPointDatabase.java
package me.robinvdb.runetracker.database;

import me.robinvdb.runetracker.domain.DataPoint;

import java.util.Collection;
import java.util.Optional;
import java.util.stream.Stream;

public interface DataPointDatabase {
    void add(DataPoint dataPoint);
    void addAll(Collection<DataPoint> dataPoints);
    void delete(DataPoint dataPoint);
    Optional<DataPoint> get(Integer id);
    Stream<DataPoint> all();
}

File: web/src/main/java/me/robinvdb/runetracker/database/DataPointDatabase.kt
package me.robinvdb.runetracker.database

import me.robinvdb.runetracker.domain.DataPoint
import java.util.Optional
import java.util.stream.Stream

interface DataPointDatabase {
    fun add(dataPoint: DataPoint)
    fun addAll(dataPoints: Collection<DataPoint>)
    fun delete(dataPoint: DataPoint)
    operator fun get(id: Int?): Optional<DataPoint>
    fun all(): Stream<DataPoint>
}
